app/views/shared/provider/_header.html.slim

```diff
@@ -1,53 +1,42 @@
+= javascript_pack_tag 'header'
 = javascript_pack_tag 'navigation'
 
-header.Header id="user_widget" class=('Header--master' if current_account.master?)
-
-  = link_to provider_admin_dashboard_path, class: 'Header-link', title: 'Dashboard' do
-      .Header-logo class=('Header-logo--withIcon' if active_menu?(:main_menu, :dashboard))
-        | Red Hat 3scale API Management
-
-  = render 'shared/api_selector'
-
-  .PopNavigation.PopNavigation--session
-    a.PopNavigation-trigger.u-toggler class=('is-current' if active_menu?(:topmenu, :help)) href="#session-menu" title="Session"
-      => icon('user')
-      => icon('bolt') if impersonating?
-    ul#session-menu.PopNavigation-list.u-toggleable
-      li.PopNavigation-listItem
-        p.PopNavigation-info
-          - if impersonating?
-            => icon('bolt')
-            ' Impersonating a virtual admin user from
-          - else
-            ' Signed in to
-          => current_user.account.name
-          ' as
-          = current_user.display_name
-          | .
-
-        = fixed_width_icon_link_to 'Sign Out', 'times', provider_logout_path, class: 'PopNavigation-link', title: "Sign Out #{current_user.username}"
-
-
-  .PopNavigation.PopNavigation--docs
-    a.PopNavigation-trigger.u-toggler class=('is-current' if active_menu?(:topmenu, :help)) href="#documentation-menu" title="Documentation"
-      => icon('question-circle')
-    ul#documentation-menu.PopNavigation-list.u-toggleable
-      li.PopNavigation-listItem
-        = fixed_width_icon_link_to 'Customer Portal', 'external-link', '//access.redhat.com/products/red-hat-3scale', class: 'PopNavigation-link', target: '_blank'
-      li.PopNavigation-listItem
-        = fixed_width_icon_link_to '3scale API Docs', 'puzzle-piece', provider_admin_api_docs_path, class: 'PopNavigation-link'
-      li.PopNavigation-listItem
-        = fixed_width_icon_link_to 'Liquid Reference', 'code', provider_admin_liquid_docs_path, class: 'PopNavigation-link'
-      - if saas?
-        li.PopNavigation-listItem
-          = fixed_width_icon_link_to "What's new?", 'leaf', '//access.redhat.com/articles/3107441#newfeaturesenhancements', class: 'PopNavigation-link', target: '_blank'
-
-  .PopNavigation.PopNavigation--account
-    - link = can?(:manage, current_account) ? provider_admin_account_path : edit_provider_admin_user_personal_details_path
-
-    a.PopNavigation-trigger class=('is-current' if active_menu?(:topmenu, :account)) href="#{link}"  title="Account Settings"
-      => icon('cog')
-
-  // HACK of the month. See features/support/current_user.rb if you are hungry for reasons
-  div class="username" style="display:none;"
-    = current_user.username
+- class_header = current_account.master ? 'Header Header--master' : 'Header'
```

Comment:
I guess it would be cleaner if we can get all these settings/options already digested from the controller with a presenter/decorator. The partial is polluted with loads of variables

Reply:
I agree, it'd be nice to get all data in a single object, but in the meantime I think we need to handle it this way